src/report/metrics.js

```diff
@@ -0,0 +1,113 @@
+const os = require("os");
+const chalk = require("chalk");
+
+const DECIMAL_BASE = 10;
+const METRIC_WIDTH = 4;
+const INSTABILITY_DECIMALS = 2;
+const YADDUM = DECIMAL_BASE ** INSTABILITY_DECIMALS;
+const COMPONENT_HEADER = "name";
+
+function getHeader(pMaxNameWidth) {
+  return `${COMPONENT_HEADER.padEnd(pMaxNameWidth)} ${"N".padStart(
+    METRIC_WIDTH + 1
+  )} ${"Ca".padStart(METRIC_WIDTH + 1)} ${"Ce".padStart(
+    METRIC_WIDTH + 1
+  )}  ${"I".padEnd(METRIC_WIDTH + 1)}`;
+}
+
+function getDemarcationLine(pMaxNameWidth) {
+  return `${"-".repeat(pMaxNameWidth)} ${"-".repeat(
+    METRIC_WIDTH + 1
+  )} ${"-".repeat(METRIC_WIDTH + 1)} ${"-".repeat(
+    METRIC_WIDTH + 1
+  )}  ${"-".repeat(METRIC_WIDTH + 1)}`;
+}
+
+function getMetricsTable(pMetrics, pMaxNameWidth) {
+  return pMetrics.map((pMetric) => {
+    return `${pMetric.name.padEnd(pMaxNameWidth, " ")}  ${pMetric.moduleCount
+      .toString(DECIMAL_BASE)
+      .padStart(METRIC_WIDTH)}  ${pMetric.afferentCouplings
+      .toString(DECIMAL_BASE)
+      .padStart(METRIC_WIDTH)}  ${pMetric.efferentCouplings
+      .toString(DECIMAL_BASE)
+      .padStart(METRIC_WIDTH)}  ${(
+      Math.round(YADDUM * pMetric.instability) / YADDUM
+    )
+      .toString(DECIMAL_BASE)
+      .padEnd(METRIC_WIDTH)}`;
+  });
+}
+
+function metricifyModule({ source, dependents, dependencies, instability }) {
+  return {
+    name: source,
+    moduleCount: 1,
+    afferentCouplings: dependents.length,
+    efferentCouplings: dependencies.length,
+    instability,
+  };
+}
+
+function metricsAreCalculable(pModule) {
```

Comment:
Maybe modules for which we don't calculate metrics shouldn't have an `instability` metric at all. If we need one for them later it's easy to derive (from our point of view instability is 0 always).

Reply:
check